solidity/contracts/wrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -96,4 +119,47 @@ contract RBTCWrapperProxy{
         return reserveAmount;
     }
     
+    /**
+     * @notice
+     * Before calling this function to swap DoC to RBTC, users need approve this contract to be able to spend or transfer their DoC tokens
+     * 
+     * @param _path         conversion path between two tokens in the network
+     * @param _amount       amount to convert from, in the source token
+     * @param _minReturn    if the conversion results in an amount smaller than the minimum return - it is cancelled, must be greater than zero
+     * 
+     * @return amount of tokens received from the conversion
+     */
+    function convertByPath(IERC20Token[] memory _path, uint256 _amount, uint256 _minReturn) public payable returns(uint256) {
+        if (msg.value != 0) {
+            require(_amount == msg.value, "The provided amount should be identical to msg.value");
+
+            (bool successOfDeposit, ) = wrbtcTokenAddress.call.value(_amount)(abi.encodeWithSignature("deposit()"));
+            require(successOfDeposit);
+
+            bool successOfApprove = wrbtc.approve(sovrynSwapNetworkAddress, _amount);
+            require(successOfApprove);
+
+            uint256 amount = sovrynSwapNetwork.convertByPath(_path, _amount, _minReturn, msg.sender, address(0), 0);
+
+            return amount;
+        }
+        else {
+            if (_path[0] == docToken) {
+                bool successOfTransferFrom = docToken.transferFrom(msg.sender, address(this), _amount);
+                require(successOfTransferFrom);
+
+                bool successOfApprove = docToken.approve(sovrynSwapNetworkAddress, _amount);
+                require(successOfApprove);
+
+                uint256 amount = sovrynSwapNetwork.convertByPath(_path, _amount, _minReturn, address(this), address(0), 0);
+
+                wrbtc.withdraw(amount);
+
+                msg.sender.transfer(amount);
+
+                return amount;
+            }
```

Comment:
Retransfer

Reply:
Can u pls explain what u mean or share more ideas?

Reply:
Can u pls explain what u mean or share more ideas? Where has the retransferring issue?